mycore-base/src/main/java/org/mycore/access/MCRAccessManager.java

```diff
@@ -426,33 +415,12 @@ public static boolean hasRule(String id, String permission) {
         }
     }
 
-    public static CompletableFuture<Boolean> checkPermission(MCRUserInformation user, Supplier<Boolean> checkSuplier) {
-        return checkPermission(user, checkSuplier, EXECUTOR_SERVICE);
-    }
-
-    public static CompletableFuture<Boolean> checkPermission(MCRUserInformation user, Supplier<Boolean> checkSuplier,
-        ExecutorService es) {
-        return CompletableFuture.supplyAsync(getWrappedFixedUserCallable(user, checkSuplier), es);
-    }
-
-    private static Supplier<Boolean> getWrappedFixedUserCallable(MCRUserInformation user,
-        Supplier<Boolean> checkSuplier) {
-        Supplier<Boolean> check = () -> {
-            try {
-                return checkSuplier.get();
-            } finally {
-                MCREntityManagerProvider.getCurrentEntityManager().clear();
-            }
-        };
-        MCRFixedUserCallable<Boolean> mcrFixedUserCallable = new MCRFixedUserCallable<>(check::get, user);
-        return () -> {
-            try {
-                return mcrFixedUserCallable.call();
-            } catch (Exception e) {
-                LOGGER.error("Exception while running ACL check for user: {}", user.getUserID(), e);
-                return false;
-            }
-        };
+    public static boolean checkPermission(MCRUserInformation user, Supplier<Boolean> checkSuplier) {
+        if (!MCRSessionMgr.hasCurrentSession()
+            || !(MCRSessionMgr.getCurrentSession() instanceof MCRScopedSession session)) {
+            throw new IllegalStateException("require an instance of MCRScopedSession");
+        }
+        return session.doAs(new MCRScopedSession.ScopedValues(user, "0.0.0.0"), checkSuplier);
```

Comment:
Don't use magic string. Add getDefaultIP() to ScopedValues().

Reply:
fixed in https://github.com/MyCoRe-Org/mycore/pull/2162/commits/231740ee2ac7ce66f4c60aa6131445f534ff1486